patches/net/minecraft/world/entity/player/Inventory.java.patch

```diff
@@ -19,14 +19,22 @@
              if (p_36049_.hasTag()) {
                  itemstack.setTag(p_36049_.getTag().copy());
              }
-@@ -237,6 +_,7 @@
+@@ -230,10 +_,14 @@
+     }
+ 
+     public void tick() {
++        int slot = 0;
+         for(NonNullList<ItemStack> nonnulllist : this.compartments) {
+             for(int i = 0; i < nonnulllist.size(); ++i) {
+                 if (!nonnulllist.get(i).isEmpty()) {
+-                    nonnulllist.get(i).inventoryTick(this.player.level(), this.player, i, this.selected == i);
++                    // Neo: Fix the slot param to be the global index instead of the per-compartment index.
++                    // Neo: Fix the selected param to only be true for hotbar slots.
```

Comment:
`this.selected` can only be between `0` and `9` so `this.selected == i` should already be correct.